### PR TITLE
Refactor integration tests to use pageId rather than page title

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR refactors the integrations tests (before we get too far with them!) so that the page assertion is based on pageId rather than page title. We did this in SLM, and the thinking was that asserting the current page on page title was fragile as content is likely to change

### CODE REVIEW

File: integration_tests/pages/overview/OverviewPage.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import Page, { PageElement } from '../page'
+
+export default class OverviewPage extends Page {
```
Comment (nathanrussell-moj-digital) [inline integration_tests/pages/overview/OverviewPage.ts:3]: We have no integration tests using this yet, but I've created it ready 👍

File: integration_tests/pages/page.ts [hunk 2]
```diff
@@ -55,4 +55,9 @@ export default abstract class Page {
     cy.get(`#${field}-error`).should('exist')
     return this
   }
+
+  hasMainHeading(expectedHeading: string) {
```
Comment (nathanrussell-moj-digital) [inline integration_tests/pages/page.ts:59]: If the test does care about page title we can call this assertion method

File: server/views/partials/layout.njk [hunk 3]
```diff
@@ -26,6 +27,25 @@
 {% endblock %}
 
 {% block bodyStart %}
+  <span class="govuk-visually-hidden" id="pageId" data-qa="{{ pageId }}"></span>
```
Comment (nathanrussell-moj-digital) [inline server/views/partials/layout.njk:30]: This is where the pageId is added to the page, in a hidden span

File: integration_tests/pages/page.ts [hunk 4]
```diff
@@ -19,7 +19,7 @@ export default abstract class Page {
   }
 
   checkOnPage(): void {
-    cy.get('h1').contains(this.title)
+    cy.get('#pageId').should('have.attr', 'data-qa').should('equal', this.pageId)
```
Comment (mattwills-moj-digital) [inline integration_tests/pages/page.ts:22]: 👍

**Review by mattwills-moj-digital — APPROVED** [review]
GS

**Pull request merged**